File: ruby/grade-school/grade_school.rb
class School
  attr_reader :roster

  def initialize
    @roster = Hash.new {|hsh, key| hsh[key] = [] }
  end

  def students(grade)
    roster[grade].sort
  end

  def add(name, grade)
    roster[grade] << name
  end

  def students_by_grade
    formatted_students = roster.sort.map do |grade, students|
      {:grade=>grade, :students=>students.sort}
    end
    formatted_students
  end

end

module BookKeeping
  VERSION = 3
end

File: ecmascript/grade-school/grade-school.js
class School {

  constructor(){
    this.students = {}
  }

  roster() {
    return this.students
  }

  add(name, grade) {
    if (this.roster().hasOwnProperty(grade)) {
      this.roster()[grade].push(name)
    }
    else {
      return this.roster()[grade] = [name]
    }
    this.roster()[grade].sort()
  }

  grade(grade) {
    if (this.roster().hasOwnProperty(grade)) {
      return this.roster()[grade]
    }
    else {
      return []
    }
  }
}

export default School
